cli/medperf/comms/entity_resources/resources.py

```diff
@@ -14,9 +14,46 @@
 
 import shutil
 import os
+import logging
+import yaml
 import medperf.config as config
 from medperf.utils import generate_tmp_path, get_cube_image_name, remove_path, untar
-from .utils import download_resource
+from .utils import download_resource, verify_or_get_hash
+
+
+def _should_get_cube(output_path, expected_hash):
+    if os.path.exists(output_path) and expected_hash:
+        if verify_or_get_hash(output_path, expected_hash):
```

Comment:
```suggestion
        if expected_hash != get_file_hash(output_path):
```
Removing `verify_or_get_hash` with its complex logic, making debugging simpler

Reply:
I do prefer splitting that logic, @hasan7n would something like this cause issues?